.eslintignore

```diff
@@ -0,0 +1,3 @@
+src/core/services/compiler/compiler.spec.js
```

Comment:
you could probably do these with the 'overrides' feature.

Reply:
I tried to add them to an `excludedFiles` array, but that didn't work. Most of them cause ESLint to actually throw a parsing error if the file is evaluated at all. Adding them to the ignore file avoided this issue.

Reply:
`test/` doesn't seem to have that much wrong with it. I suspect they can be added to the `overrides.2.files` array.

Reply:
I started into cleaning those up and getting them passing, but there were a number of things that I am not comfortable changing at this time. I'd like to keep that as part of a separate PR since those files are actually used to test the library and changing them would make this PR no longer 'merge safe'.